js/src/miscellaneous.js

```diff
@@ -83,12 +83,11 @@ module.exports = (common) => {
     })
 
     // must be last test to run
-    it('.stop', (done) => {
+    it('.stop', function (done) {
+      this.timeout(10 * 1000)
       // TODO: go-ipfs returns an error, https://github.com/ipfs/go-ipfs/issues/4078
       ipfs.stop((err) => {
-        if (err && err.message !== 'read ECONNRESET') {
-          expect(err).to.not.exist()
-        }
+        // Retry
         ipfs.stop((err) => {
           expect(err).to.exist()
```

Comment:
error should not exist.

Reply:
Gotcha, then we need to check for `go` and then expect an error

```js
if (!withGo)
   expect(err).to.not.exist()
```

Here's an [example](https://github.com/ipfs/interface-ipfs-core/blob/c4934ca0b3b43f5bfc1ff5dd38f85d945d3244de/js/src/stats.js#L23) of detecting `go-ipfs`.